consensus/ibft/signer/signer.go

```diff
@@ -0,0 +1,336 @@
+package signer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/0xPolygon/polygon-edge/validators"
+)
+
+var (
+	ErrEmptyCommittedSeals        = errors.New("empty committed seals")
+	ErrEmptyParentCommittedSeals  = errors.New("empty parent committed seals")
+	ErrInvalidCommittedSealLength = errors.New("invalid committed seal length")
+	ErrInvalidCommittedSealType   = errors.New("invalid committed seal type")
+	ErrRepeatedCommittedSeal      = errors.New("repeated seal in committed seals")
+	ErrNonValidatorCommittedSeal  = errors.New("found committed seal signed by non validator")
+	ErrNotEnoughCommittedSeals    = errors.New("not enough seals to seal block")
```

Comment:
Do we really need to define 7 errors instead of 1?

Reply:
Why do you think so? Basically these errors are thrown from different methods and parts.

Reply:
What do you think about using a single error?

Why do you need to have different ones, if the committed seal is invalid, all the same?

Reply:
Personally, I don't agree with @zivkovicmilos. Imagine that you need to debug an issue with a committed seal and the client sends you a log with an error **Invalid committed seal**. I would be really pissed if i had to debug where out of 10 places error occured. 
The more descriptive errors we have the better.

Reply:
@zivkovicmilos ping